vertx-sql-client/src/main/java/io/vertx/sqlclient/data/NullValue.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.data;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.math.BigDecimal;
+import java.time.*;
+import java.time.temporal.Temporal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Instances of this class indicate the type associated with a {@code NULL} column value.
+ * This is useful when infering database types from {@link io.vertx.sqlclient.Tuple} items.
+ */
+public class NullValue {
+
+  private static final Map<Class<?>, NullValue> cached = new HashMap<>();
```

Comment:
we need to avoid caching such values, users should cache themselves the null value instance (we can document that)

Reply:
We don't cache other `NullValue` instances than those defined in this class (except the `Buffer`/`Temporal` and their array equivalents). Perhaps the name `cache` is confusing and you'd prefer `lookupTable`?

Reply:
right sorry for that, then I think we could just have a comment above the cache saying that we only cache JVM types (we can also cache Buffer I think).

besides and that's minor we should avoid caching in the constructor and instead it should be done after the object has been fully initialized (so using a static private method).